iOS/DuckDuckGo/TabViewController.swift

```diff
@@ -2471,9 +2471,20 @@ extension TabViewController {
 
         if case .blob = SchemeHandler.schemeType(for: url) {
             return (.download, nil)
-        } else if let download = downloadManager.makeDownload(navigationResponse: navigationResponse, cookieStore: cookieStore) {
-            downloadManager.startDownload(download)
-            return (.cancel, download)
+        } else {
+            do {
+                if let download = try downloadManager.makeDownload(navigationResponse: navigationResponse, cookieStore: cookieStore) {
```

Comment:
Called after the user confirms "Save to Downloads" or for auto-preview. On failure, log and show display and error via ActionMessageView since the user initiated the action.